conda_smithy/lint_recipe.py

```diff
@@ -886,6 +887,125 @@ def check_pins_build_and_requirements(top_level):
         if osx_hint not in hints:
             hints.append(osx_hint)
 
+    # stdlib issues in CBC
+    cbc_lines = []
+    if conda_build_config_filename:
+        with open(conda_build_config_filename, "r") as fh:
+            cbc_lines = fh.readlines()
+
+    # filter on osx-relevant lines
+    pat = re.compile(
+        r"^([^\#]*?)\s+\#\s\[.*(not\s(osx|unix)|(?<!not\s)(linux|win)).*\]\s*$"
+    )
+    # remove lines with selectors that don't apply to osx, i.e. if they contain
+    # "not osx", "not unix", "linux" or "win"; this also removes trailing newlines
+    cbc_lines_osx = [pat.sub("", x) for x in cbc_lines]
+    cbc_content_osx = "\n".join(cbc_lines_osx)
+    cbc_osx = get_yaml().load(cbc_content_osx) or {}
+    # filter None values out of cbc_osx dict, can appear for example with
+    # ```
+    # c_stdlib_version:  # [unix]
+    #   - 2.17           # [linux]
+    #   # note lack of osx
+    # ```
+    cbc_osx = dict(filter(lambda item: item[1] is not None, cbc_osx.items()))
+
+    # for fallback, assume ordering [x64, arm64] which is used (almost) universally
+    baseline_version = ["10.13", "11.0"]
```

Comment:
We could add _yet another_ lint if the order here is "wrong" (from the POV of the rest of the processing), i.e. for
```
c_stdlib_version:           # [osx]
  - 11.0                    # [osx and arm64]
  - 10.13                   # [osx and x86_64]
```

Reply:
Or alternatively, if we find `version(first_entry) > version(second_entry)`, we assume that we've got arm64 sorted before, and just reverse the list. That would still leave an exceedingly small corner case open[^1], but that shouldn't be an issue IMO

[^1]: if osx-arm64 genuinely uses a lower deployment-target or sdk than osx-64, which I don't think ever makes sense

Reply:
Why does the ordering matter here? Is there a zip I am not remembering?

Reply:
Mostly the whole thing is about having 10.13 somewhere. I added 11.0 for osx-arm64 as well because otherwise I cannot diagnose a handful of cases.

However the tradeoff is the need to assume some order between osx-64 and osx-arm64 (or do a bunch more regex work, which currently just filters out anything that's not explicitly osx-related).

Reply:
> Or alternatively, if we find `version(first_entry) > version(second_entry)`, we assume that we've got arm64 sorted before, and just reverse the list.

I ended up going with that. This should have essentially zero real failure modes AFAICT. 🤞